.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -3,6 +3,8 @@
 ## PR Checklist
 
 <!-- Please mark any checkboxes that do not apply to this PR as [N/A]. -->
+**Linked Issue**
+- [ ] Mentioned issue that it closes using a phrase like "closes #123" to link it back to the issue.
```

Comment:
I'm pretty sure it can be mentioned anywhere and that should be fine, no?

Reply:
I'd say that it is better to have it in the PR than in the commit message. See https://github.com/matplotlib/matplotlib/issues/21506 for a scary example of what happens to the issue thread every time someone (force-)pushes a new version...

Reply:
Yes, while it works to put it in a commit message, that gets a bit more messy, and so if we are telling people where to do it, we should say in the PR description.

> You can also use closing keywords in a commit message. The issue will be closed when you merge the commit into the default branch, but the pull request that contains the commit will not be listed as a linked pull request.

(from https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
Because it does not link in the issue, it can be harder to track commit message versions.

Note that it _does not_ work in a comment, you must edit the main description to change it later.

Also note that putting "closes #123" in the template, will, undoubtedly, cause every PR to link to that (closed) PR, which is probably undesirable... (It may also cause notifications for that PR on merges... I don't think it does for just linking, not sure what happens if it is already closed...)

Perhaps "closes #ABCDE" instead...

Reply:
Or "closes #000", as that won't link to anything, but is at least numeric (note `#0123` still links to #0123, so actually does have to be all zeros for that format)

Reply:
And also note that it should not be in the PR _title_ as sometimes happens.

Reply:
I think #000 would be better as it is still numeric.